In Gentoo, QScintilla for Qt5 has the same library name
as its Qt4 variant.

--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -1460,7 +1460,7 @@
     ;;
     5)
       QT_MODULES="Qt5Core Qt5Gui Qt5Network Qt5OpenGL Qt5PrintSupport"
-      octave_qscintilla_libnames="qscintilla2-qt5 qscintilla2_qt5 qt5scintilla2"
+      octave_qscintilla_libnames="qscintilla2_qt5 qscintilla2"
     ;;
     *)
       AC_MSG_ERROR([Unrecognized Qt version $qt_version])
